NRAS.py

```diff
@@ -0,0 +1,101 @@
+from abc import ABC, abstractmethod
+
+
+class BiologicalSequence(ABC):
+
+    @abstractmethod
+    def __getitem__(self, index) -> str:
+        pass
+
+    @abstractmethod
+    def __str__(self) -> str:
+        pass
+
+    @abstractmethod
+    def __repr__(self) -> str:
+        pass
+
+    @abstractmethod
+    def check_alphabet(self) -> bool:
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    def __init__(self, seq: str):
+        self.seq = seq
+        self.length = len(self.seq)
+        self.check_alphabet()
+
+    def __getitem__(self, slc) -> str:
+        if isinstance(slc, int):
+            return self.seq.__getitem__(slc - 1)
+        elif isinstance(slc, slice):
+            new_slice = slice(slc.start - 1, slc.stop, slc.step)
+            return self.seq.__getitem__(new_slice)
```

Comment:
В идеале у нас должен возвращаться тот же класс(ДНК, РНК, аминокислота и т.п.), так что неплохо было бы добавить преобразование полученной строки + можно обойтись без условий, так как \_\_getitem__ у seq (является экземпляром str) имеет весь необходимый функционал)

```suggestion
    def __getitem__(self, index) -> "Self": # для такой типизации нужен from typing import Self
        return self.__class__(self.seq[index]) 
```